derive_builder/src/options/darling_opts.rs

```diff
@@ -319,6 +319,16 @@ impl Options {
             .fields
     }
 
+    /// A builder requires `Clone` to be derived if any of its fields use the mutable or immutable pattern.
+    ///
+    /// This can technically be expressed two ways:
+    ///
+    /// 1. The caller chooses the `owned` pattern at the struct level
+    /// 1. The caller chooses the `owned` pattern on every field
+    pub fn requires_clone(&self) -> bool {
+        self.fields().any(|f| f.pattern().requires_clone())
```

Comment:
@azriel91 if there's a place where we should check the struct's pattern, it'd be here. However, I don't think we strictly _need_ to; unless I'm missing something, it only matters what the individual fields are doing: A mutable zero-field builder shouldn't need `Clone` to operate.

Reply:
True that (though one would wonder why you'd have a zero field builder :stuck_out_tongue:)

Reply:
Wait, I forgot that `build` takes a `&self` if the struct-level one isn't marked `owned`. I've updated the logic accordingly.